.appveyor.yml

```diff
@@ -0,0 +1,117 @@
+#cache:
+#    - '%LOCALAPPDATA%\Composer\files -> composer.lock'
+init:
+  - SET PATH=C:\Program Files\OpenSSL;c:\tools\php;%PATH%
+  - SET COMPOSER_NO_INTERACTION=1
+  - SET PHP=1
+  - SET ANSICON=121x90 (121x90)
+services:
+#  - mssql2014
+#  - mysql
+#  - postgresql94
+
+## Install PHP and composer, and run the appropriate composer command
+install:
+    - IF EXIST c:\tools\php (SET PHP=0)
+    - ps: appveyor-retry cinst --ignore-checksums -y php --version ((choco search php --exact --all-versions -r | select-string -pattern $Env:php_ver_target | Select-Object -first 1) -replace '[php|]','')
+    - cinst -y sqlite
+    - ps: If ($env:db -like "mariadb") {cinst -y mariadb}
+    - cd c:\tools\php
+    #get the MSSQL DLL's
+    - ps: >-
+        If ($env:php_ver_target -eq "7.0") {
+          If ($env:PHP -eq "1") {
+            If ($env:db -like "mssql") {
+              appveyor DownloadFile https://github.com/Microsoft/msphpsql/releases/download/4.1.5-Windows/7.0.zip
+              7z x 7.0.zip > $null
+              copy 7.0\x64\php_pdo_sqlsrv_7_nts.dll ext\php_pdo_sqlsrv_nts.dll
+              copy 7.0\x64\php_sqlsrv_7_nts.dll ext\php_sqlsrv_nts.dll
+              Remove-Item c:\tools\php\* -include .zip}}}
+    - ps: >-
+        If ($env:php_ver_target -eq "7.1") {
+          If ($env:db -like "mssql") {
+            If ($env:PHP -eq "1") {
+              appveyor DownloadFile https://github.com/Microsoft/msphpsql/releases/download/4.1.5-Windows/7.1.zip
+              7z x 7.1.zip > $null
+              copy 7.1\x64\php_pdo_sqlsrv_71_nts.dll ext\php_pdo_sqlsrv_nts.dll
+              copy 7.1\x64\php_sqlsrv_71_nts.dll ext\php_sqlsrv_nts.dll
+              Remove-Item c:\tools\php\* -include .zip}}}
+    #Set PHP environment
+    - IF %PHP%==1 copy php.ini-production php.ini /Y
+    - IF %PHP%==1 echo date.timezone="UTC" >> php.ini
+    - IF %PHP%==1 echo extension_dir=ext >> php.ini
+    - IF %PHP%==1 echo extension=php_openssl.dll >> php.ini
+    - IF %PHP%==1 echo extension=php_mbstring.dll >> php.ini
+    - IF %PHP%==1 echo extension=php_fileinfo.dll >> php.ini
+    - IF %PHP%==1 echo extension=php_pdo_mysql.dll >> php.ini
+    - ps: >-
+        If ($env:db -like "mssql") {
+        If ($env:PHP -eq "1") {
+          echo extension=php_pdo_sqlsrv_nts.dll >> php.ini}
+        If ($env:PHP -eq "1") {
+          echo extension=php_sqlsrv_nts.dll >> php.ini}}
+    - IF %PHP%==1 echo extension=php_pdo_pgsql.dll >> php.ini
+    - IF %PHP%==1 echo extension=php_pdo_sqlite.dll >> php.ini
+    - IF %PHP%==1 echo extension=php_sqlite3.dll >> php.ini
+    - IF %PHP%==1 echo extension=php_mysqli.dll >> php.ini
+    - IF %PHP%==1 echo extension=php_pgsql.dll >> php.ini
+    - IF %PHP%==1 echo @php %%~dp0composer.phar %%* > composer.bat
+    - appveyor-retry appveyor DownloadFile https://getcomposer.org/composer.phar
+    - cd c:\projects\dbal
+    - IF %dependencies%==lowest appveyor-retry composer update --prefer-lowest --no-progress --profile -n
```

Comment:
If seen it in a few places where it's used in the "allowed failures" to get an idea of what's coming next and if the next release of a dependency breaks a package. 

There are also times where it's used to allow for testing when the consuming repos might have higher or lower requirements and the package itself is typically happy in any case with a broader range of dependency versions (put they want to test for it when changes are made).